src/main/scala/com/gu/fastly/CrierEventProcessor.scala

```diff
@@ -3,22 +3,32 @@ package com.gu.fastly
 import com.amazonaws.services.kinesis.model.Record
 import com.gu.crier.model.event.v1.Event
 import com.gu.thrift.serializer.ThriftDeserializer
-import scala.util.Try
+
+import scala.util.{Failure, Success, Try}
 
 object CrierEventProcessor {
 
   def process(records: Seq[Record])(purge: Event => Boolean) = {
-    val processingResults: Iterable[Boolean] = records.flatMap { record =>
-      val event = eventFromRecord(record)
-      event.map { e =>
-        purge(e)
-      }.recover {
-        case error =>
-          println("Failed to deserialize Crier event from Kinesis record. Skipping.")
-          false
-      }.toOption
+    val crierEvents = records.flatMap { record =>
+      eventFromRecord(record) match {
+        case Success(event) =>
+          Some(event)
+        case Failure(error) =>
+          println("Failed to deserialize Crier event from Kinesis record. Skipping: " + error.getMessage)
+          None
+      }
+    }
+
+    // During initial testing of model update fail the lambda if there are deserialization failures.
+    if (crierEvents.size < records.size) {
```

Comment:
Just to check, if this exception is thrown, does this mean the current batch of records will be retried (meaning we don't lose any records that we failed to deserialise)?

Reply:
Discussed on call. Intentional jamming of the watermark so that failures would replay on rollback.